### PR TITLE
feat(private-database): add CPU throttle to CloudDB stats

### PR DESCRIPTION
**Pull request by wdhif**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #PUDBLOW-1532
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Adds a CPU saturation section, with the percentage spend in throttle state and a modal with the list of CPU saturation.

Without CPU saturation:
Capture d’écran 2022-04-13 à 17 12 06
Capture d’écran 2022-04-13 à 17 12 16

With CPU saturation:
Capture d’écran 2022-04-13 à 17 12 47
Capture d’écran 2022-04-13 à 17 12 56

### :flags: Translations

- [x] TRANS-43240

### CODE REVIEW

**Review by mdavranche — DISMISSED** [review]

**Review by carmelo42 — DISMISSED** [review]

File: packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.controller.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import clone from 'lodash/clone';
+
+angular.module('App').controller(
+  'PrivateDatabaseCpuThrottleCtrl',
+  class PrivateDatabaseCpuThrottleCtrl {
```
Comment (marie-j) [inline packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.controller.js:5]: Not possible to use `export` syntax here ?

File: packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<div data-ng-controller="PrivateDatabaseCpuThrottleCtrl as cpuThrottleCtrl">
+    <div
```
Comment (marie-j) [inline packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.html:2]: current practice is to use a specific `modal` layout as wizard is deprecated.
Also are we sure about datagrid display in a modal ?

File: packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.controller.js (hunk 1 above)
Comment (wdhif) [reply packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.controller.js:5]: This code is actually made to be iso with the code that is managing the OOM. I would prefer to keep the codebase coherent.

File: packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.html (hunk 2 above)
Comment (wdhif) [reply packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.html:2]: As with the controller, I am doing this to keep the codebase consistent.

File: packages/manager/apps/web/client/app/private-database/dashboard/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -458,6 +460,16 @@
...
+  "privateDatabase_modale_cpu_throttle_title": "Saturations CPU",
+  "privateDatabase_modale_cpu_throttle_state_button": "Vérifier les saturations CPU",
+  "privateDatabase_modale_cpu_throttle_close_button": "Fermer",
+  "privateDatabase_modale_cpu_throttle_info": "Saturations CPU survenues ces {{t0}} derniers jours.<br />Seules les {{t1}} dernières saturations sont disponibles.",
+  "privateDatabase_modale_cpu_throttle_tab_event_start_date": "Date de début",
+  "privateDatabase_modale_cpu_throttle_tab_event_end_date": "Date de fin",
+  "privateDatabase_modale_cpu_throttle_tab_duration": "Durée de la saturation",
+  "privateDatabase_modale_cpu_throttle_ongoing_cpu_throttle": "Saturation CPU en cours",
+  "privateDatabase_modale_cpu_throttle_no_cpu_throttle": "Pas de saturation CPU.",
+  "privateDatabase_modale_cpu_throttle_info_order": "Nous vous conseillons d'améliorer vos requêtes et index ou de répartir vos bases de données sur plus d'instances CloudDB pour pallier à ces problèmes.",
```
Comment (antleblanc) [inline packages/manager/apps/web/client/app/private-database/dashboard/translations/Messages_fr_FR.json:472]:
```suggestion
  "privateDatabase_modale_cpu_throttle_info_order": "Nous vous conseillons d'améliorer vos requêtes et index ou de répartir vos bases de données sur plus d'instances CloudDB pour pallier ces problèmes.",
```

**Review by antleblanc — CHANGES_REQUESTED** [review]

File: packages/manager/apps/web/client/app/private-database/dashboard/translations/Messages_fr_FR.json (hunk 3 above)
Comment (Meuh-42) [reply packages/manager/apps/web/client/app/private-database/dashboard/translations/Messages_fr_FR.json]: Suggestion applied

File: packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.controller.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    constructor(
+      $scope,
+      $q,
+      $stateParams,
+      $translate,
+      Alerter,
+      CpuThrottleService,
+      PrivateDatabase,
+    ) {
```
[the remark this answers is not in the document]
Comment (Meuh-42) [reply packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.controller.js]: Suggestion applied

File: packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.html [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                    data-title="'privateDatabase_modale_cpu_throttle_tab_event_end_date' | translate"
+                >
+                    <span
+                        data-ng-if="$row.endDate !== null"
+                        data-ng-bind="$row.endDate | date:'medium'"
+                    ></span>
+                    <span class="text-center" data-ng-if="$row.endDate === null"
+                        >{{::
+                        'privateDatabase_modale_cpu_throttle_ongoing_cpu_throttle'
+                        | translate }}</span
```
[the remark this answers is not in the document]
Comment (Meuh-42) [reply packages/manager/apps/web/client/app/private-database/cpu-throttle/private-database-cpu-throttle.html]: Suggestion applied

**Review by Meuh-42 — DISMISSED** [review]

**Review by antleblanc — APPROVED** [review]

**Review by JeremyDec — DISMISSED** [review]

**Comment by mferon** [comment]
LGTM

**Review by marie-j — APPROVED** [review]

**Pull request merged**